client/state/happychat/selectors.js

```diff
@@ -97,16 +112,25 @@ export const getHappychatTimeline = createSelector(
 	state => map( state.happychat.timeline, 'id' )
 );
 
-export const canUserSendMessages = createSelector(
-	state => (
-		getHappychatConnectionStatus( state ) === 'connected' &&
-		! includes(
-			[	HAPPYCHAT_CHAT_STATUS_PENDING, HAPPYCHAT_CHAT_STATUS_MISSED,
-				HAPPYCHAT_CHAT_STATUS_ABANDONED ],
-			getHappychatChatStatus( state )
-		) && isHappychatAcceptingChats( state )
-	),
-	[ getHappychatConnectionStatus, getHappychatChatStatus ]
+/**
+ * Returns true if the user should be able to send messages to operators based on
+ * chat status. For example new chats and ongoing chats should be able to send messages,
+ * but blocked or pending chats should not.
+ * @param {Object} state - global redux state
+ * @return {Boolean} Whether the user is able to send messages
+ */
+export const canUserSendMessages = state => (
+	isHappychatAvailable( state ) &&
```

Comment:
I mentioned this on the other comment, but just to be clear — I'd advocate that tests for `canUserSendMessages` should mock the `isHappychatAvailable` selector. As long as the intent of the `isHappychatAvailable` selector never changes (indicating whether Happychat can take chats from this user) `canUserSendMessages` shouldn't care about its implementation details including whether there's a connection or not.